tests/list_test_suites.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+# List of all tests that should be executed
+tests=($1)
+
+# Count of tests
+total_tests=${#tests[@]}
+
+# Number of tests by suite
+tests_n=$(expr $total_tests / $2)
+
+# Double number (in case of remaining tests and division is not exactly)
+double=$(expr $tests_n \* 2)
+
+# divided suites
+suites=
+
+for((i=0; i < $total_tests; i+=tests_n))
+do
+  remaining=$(expr $i + $double)
+
+  # if it can't be divided exactly so the last test will include the remaining tests.
+  if [[ $remaining -lt $total_tests ]]; then
```

Comment:
Note that some tests should always be in the same suite (i.e. executed in parallel). For example, all pubsub tests should be executed in parallel to avoid overloading Service Bus

Reply:
In this case all pubsub tests should be executed sequentially rather than parallel, is that you meant?

If yes, indeed, as you can see the test suite division is not explicit, I was wondering if it really needs to be. Thoughts?

An option might be prefixing the folder name with the desired test suite, something like:

1-pubsub_test_b
1-pubsub_test_a

Folders that not contains the prefix wouldn't have any restriction, how about that?

Another option is having subfolders

/pubsub/test_a
/pubsub/test_b

Reply:
Yes, sorry, I meant sequentially.

Any implementation would be fine.